internal/controller/monovertexrollout/monovertexrollout_controller.go

```diff
@@ -584,38 +574,29 @@ func (r *MonoVertexRolloutReconciler) IncrementChildCount(ctx context.Context, r
 	return currentNameCount, nil
 }
 
-func (r *MonoVertexRolloutReconciler) ChildIsDrained(ctx context.Context, monoVertexDef *unstructured.Unstructured) (bool, error) {
-	monoVertexStatus, err := numaflowtypes.ParseMonoVertexStatus(monoVertexDef)
-	if err != nil {
-		return false, fmt.Errorf("failed to parse MonoVertex Status from MonoVertex CR: %+v, %v", monoVertexDef, err)
-	}
-	monoVertexPhase := monoVertexStatus.Phase
-
-	return monoVertexPhase == "Paused" /*&& monoVertexStatus.DrainedOnPause*/, nil // TODO: should Numaflow implement?
-}
-
-func (r *MonoVertexRolloutReconciler) Drain(ctx context.Context, monoVertexDef *unstructured.Unstructured) error {
-	patchJson := `{"spec": {"lifecycle": {"desiredPhase": "Paused"}}}`
-	return kubernetes.PatchResource(ctx, r.client, monoVertexDef, patchJson, k8stypes.MergePatchType)
+func (r *MonoVertexRolloutReconciler) Recycle(ctx context.Context,
+	monoVertexDef *unstructured.Unstructured,
+	c client.Client,
+) error {
+	return kubernetes.DeleteResource(ctx, c, monoVertexDef)
```

Comment:
replacing the original `ChildIsDrained()` and `Drain()` calls which were part of the `progressiveController` interface with `Recycle()` (since draining is not something MVX does, this is more generic)